src/include/drivers/usb/usb.h

```diff
@@ -160,6 +160,7 @@ struct usb_dev {
 	void *config_buf;
 	struct usb_desc_device dev_desc;
 	struct usb_desc_interface iface_desc;
+	struct usb_desc_configuration conf_desc;
```

Comment:
You can use the beginning of `config_buf` instead:
```
config = (struct usb_desc_configuration *) dev->config_buf;
```

Reply:
Just because `conf_desc` is not filled anywhere

Reply:
I needed conf_desc in lsusb.c for displaying configuration descriptors

Reply:
Yes, I understand, but who fills this `conf_desc`?

Reply:
Sorry, but I didnt get what do you mean by filling the struct

Reply:
You defined new field `conf_desc` in the structure. This field is uninitialized. I think you can try to use `lsusb -v` on https://github.com/embox/embox/wiki/Using-USB-on-qemu as I proposed below, and see what will happen.

Reply:
Thank you, I used config_buf as you mentioned and removed conf_desc from struct usb_dev. I also put a check on config for null values as "Bus 000 Device 000 ID 0000:0000" in qemu returns null pointer. And  I tried `lsusb -v` on https://github.com/embox/embox/wiki/Using-USB-on-qemu it works.

Reply:
Now it looks good, thank you.